src/api/ApiBuilder.js

```diff
@@ -1,51 +1,101 @@
 import Vue from 'vue'
 import Request from '@/api/Request'
 
-const build = (config, environment) => {
-  return Object.keys(config).reduce((result, key) => {
-    const { resources, ...props } = config[key]
-    result[key] = buildResources(resources, apiProps(key, props, environment))
-    return result
+/*
+  This will construct the api based on any config that has been passed
+  The api will be a nested structure
+  It would create something like { traction: { samples, saphyr: libraries, ... }, ...}
+  So traction.saphyr.libraries would be a callable Request
+  See Request for what it can do
+  This is a world away from the first pass. It should be a lot easier to understand.
+  There is a lot of repetition of the arguments but this makes it more readable.
+  The next stage would be to use objects for the arguments or types to remove the repetition.
+  It would also be better to have pipelines as nested resources and use recursion for better flexibility
+*/
+
+// we need to create a sub class of Vue and return the constructor
+const cmp = Vue.extend(Request)
+
+/*
+ * @param {*} config - A piece of json containing all the apis, resources and pipeline
+ * @param {*} environment - The current node environment which should contain the variables for the base URLs which are sensitive
+ * @returns {*} an object of apis
+ */
+const build = ({ config, environment }) => {
+  return config.reduce((result, api) => {
+    return {
+      ...result,
+      // each api will be a property and we need to extract the baseURL from the environment
+      [api.name]: buildApi({ ...api, baseURL: environment[`${api.baseURL}`] }),
```

Comment:
Is there a reason for the interpolation here? Would:

```js
environment[api.baseURL]
```

Do something different?

Reply:
Good point. It is probably just an artifact from the previous code.